tests/python/gaia-ui-tests/gaiatest/form_controls/header.py

```diff
@@ -10,10 +10,24 @@ class GaiaHeader(Widget):
 
     _back_button_locator = (By.CSS_SELECTOR, 'button.action-button')
     _close_button_locator = (By.CSS_SELECTOR, 'button[type="reset"]')
+    _h1_locator = (By.CSS_SELECTOR, 'h1')
 
-    def go_back(self,app=None, exit_app=False, statusbar=False):
+    @property
+    def text(self):
+        self._wait_to_be_ready()
+        return self.root_element.find_element(*self._h1_locator).text
```

Comment:
I guess if the Widget class was instantiated successfully, then it means the root_element was found already, so should be no need to switch to shadow DOM.